### PR TITLE
Adding event on Calendar

### PR DESCRIPTION
**Pull request by suave990**
# PR Description
fixing the bug on the dashboard on the calendar

Description of tasks that were expected to be completed
In case you add an event it must stay whether you refresh or login after log out.

How has this been tested?
run it locally and try to add an event with such a route /dashboard/calendar

# Number of Commits

1 Commit

# Screenshots (If appropriate)
Screenshot 2023-06-17 at 14 05 36

# Please check this Checklist before you submit your PR:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code generate no warnings
- [x] My test coverage meet the set test coverage threshold
- [x] There are no vulnerabilities
- [x] There are no conflicts with the base branch

### CODE REVIEW

**Review by ceelogre — APPROVED** [review]

**Review by UwaweTera — APPROVED** [review]

**Review by teerenzo — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -69,6 +69,7 @@
     "i18next-xhr-backend": "^3.2.2",
     "istanbul": "^0.4.5",
     "jwt-decode": "^3.1.2",
+    "moment": "^2.29.4",
```
Comment (ceelogre) [inline package.json:72]: Moment is no longer actively maintained. https://momentjs.com/docs/#/-project-status/

File: src/components/Calendar.tsx [hunk 2]
```diff
@@ -8,13 +8,16 @@ import DatePicker from 'react-datepicker';
...
 const initialData: EventInput[] = [
   {
     title: 'Event title 1',
-    start: '2022-07-21',
-    end: '2022-07-21',
+    start: '2023-06-12',
+    end: '2023-06-14',
     hostName: 'Samuel NISHIMWE',
     timeToStart: '14:00',
     timeToFinish: '16:00',
```
Comment (ceelogre) [inline src/components/Calendar.tsx]: You could have removed the initial data. It doesn't serve any purpose right now.

File: src/components/Calendar.tsx [hunk 3]
```diff
@@ -48,25 +51,58 @@ const Calendar = () => {
     timeToStart: '',
     timeToFinish: '',
   });
-  const [data, setData] = useState<EventInput[]>(initialData);
+  const [data, setData] = useState<EventInput[]>([]);
+  const [getEvents] = useLazyQuery(GET_EVENTS);
+  const [createEvent] = useMutation(ADD_EVENT);
+
+  useEffect(() => {
+    console.log(data);
```
Comment (ceelogre) [inline src/components/Calendar.tsx]: No console log on pushed  code.

**Review by ceelogre — COMMENTED** [review]
Take a  look at the comments I've made. They should all be addressed when we are re-visiting this feature.

**Pull request merged**